tests/test_lightningd.py

```diff
@@ -651,9 +654,7 @@ def test_connect_by_gossip(self):
         """
         l1 = self.node_factory.get_node()
         l2 = self.node_factory.get_node()
-        # Force l3 to give its address.
-        l3port = self.node_factory.get_next_port()
-        l3 = self.node_factory.get_node(options={"ipaddr": "127.0.0.1:{}".format(l3port)})
+        l3 = self.node_factory.get_node(options={"ipaddr": "127.0.0.1"})
```

Comment:
Is port not needed here?  The reason why `ipaddr` is specified is to make `l3` gossip its own address and port correctly, else `l1` will be unable to look up `l3` via gossip.

#1450 makes `l3` gossip correctly (currently we only gossip `--ipaddr` or what is auto-detected as our address) so the option could be removed there.

Reply:
Hmm the CI seems to work, so I guess it does work?  Hmm.